src/irmin-pack/inode.ml

```diff
@@ -1827,8 +1894,73 @@ struct
       Bin.v ~root ~hash:(lazy i.hash) v
 
     let decode_bin_length = decode_compress_length
+
+    let decode_children_offsets ~entry_of_offset ~entry_of_hash t pos_ref =
+      let i = decode_compress t pos_ref in
+      let { Compress.tv; _ } = i in
+      let v =
+        match tv with
+        | V0_stable v | V0_unstable v -> v
+        | V1_root { v; _ } | V1_nonroot { v; _ } -> v
+      in
+      let entry_of_address = function
+        | Compress.Offset offset -> entry_of_offset offset
+        | Hash h -> entry_of_hash h
```

Comment:
FYI: Reaching this line is the sign that `h` is dangling. 

In Tezos it only happens when importing a full snapshot because of a small bug in `check_protocol_commit_consistency` that was fixed since.

You can leave your code as is.